build/export-openapi.sh

```diff
@@ -84,7 +84,8 @@ for f in *.json; do
   contents=$(cat "$f" | jq 'del(.description)' | sed 's/\\n/\\\\n/g' | sed '$d' | sed '1d' | sed ':a;N;$!ba;s/\n/\\n/g' | sed 's/\$/\\$/g' | sed 's/\&/\\&/g' | sed 's@\"@\\"@g')
   ref=$(basename "$f" .json)
 
-  find -maxdepth 1 -name '*.json' | xargs sed -i 's@"$ref": "#/definitions/'"$ref"'"@'"$contents"'@g'
+  find -maxdepth 1 -name '*.json' | xargs -I {} bash -c 'sed -i "s@\"$1\": \"#/definitions/$1\"@$2@g" {}' _ "$ref" "$contents"
```

Comment:
Yes, makes sense. Thanks for pointing this out. Will look into it👍🏻

Reply:
The command using `xargs sed` without the `-i` option runs successfully, but it does not replace the contents as intended:

```
find -maxdepth 1 -name '*.json' | xargs sed 's@"$ref": "#/definitions/'"$ref"'"@'"$contents"'@g'
```

Similarly, the following commands also execute without errors but fail to perform the intended replacements:

```
1) find -maxdepth 1 -name '*.json' | xargs -I 'sed -i "s@"$ref": "#/definitions/'"$ref"'"@'"$contents"'@g"'

2) find -maxdepth 1 -name '*.json' | xargs -n1 -I 'sed -i "s@"$ref": "#/definitions/'"$ref"'"@'"$contents"'@g"'

3) find -maxdepth 1 -name '*.json' | xargs -I {} bash -c 'sed -i "s@\"$ref\": \"#/definitions/$1\"@$2@g" {}' _ "$ref" "$contents"
```

I also attempted using a hardcoded value:
```
find -maxdepth 1 -name '*.json' | xargs sed 's/"$ref"/AGONES/g'
```
same here, no error and no replacement with "AGONES".

Reply:
I tested the following command with a sample JSON file locally, and it worked, but it's not working for our use case:

```
find -maxdepth 1 -name 1.json | xargs sed -i -e '/"$ref": "#\/definitions\/'"$ref"'"/s/'"$ref"'/'"$contents"'/g'
```
I would appreciate your insights and suggestions on how to proceed further to address this issue effectively...

Reply:
Confirmed the above command still have `Argument list too long` error.

Reply:
I'm fairly sure that the issue is the `xargs` not the `sed` command, so a few options that may work/help:

1. Use `-exec` on `find`, rather than `xargs` (
1. Use `parrallel` https://tecadmin.net/solutions-to-the-argument-list-too-long-error/

Reply:
yes, I have tried with the below awk command and it is working fine with a simple file.

 `find -maxdepth 1 -name 1.json | xargs awk -i inplace -v ref='"\\$ref": "#/definitions/'"$ref"'"' -v contents="$contents" '{gsub(ref, contents)}; {print}'`

```
$ cat 1.json 
{
  "name": "Spring",
  "age": 300,
  "$ref": "#/definitions/k8s",
  "hobbies": ["AAA", "BBB", "CCC"],
  "address": {
    "street": "123 M st",
    "city": "SF",
    "state": "CA",
    "zip": "123"
  }
}

$ echo $ref
k8s

$ echo $contents
support version update

$ find -maxdepth 1 -name 1.json | xargs awk -i inplace -v ref='"\\$ref": "#/definitions/'"$ref"'"' -v contents="$contents" '{gsub(ref, contents)}; {print}'

$ cat 1.json 
{
  "name": "Spring",
  "age": 300,
  support version update,
  "hobbies": ["AAA", "BBB", "CCC"],
  "address": {
    "street": "123 M st",
    "city": "SF",
    "state": "CA",
    "zip": "123"
  }
}
```

Reply:
I think Kalai tried getting rid of `xargs` and using a loop instead, which still throw the Arguments too long error?

Reply:
Absolutely have not tried it, so may need some tweaking, but does this work:

```
find -maxdepth 1 -name '*.json' -exec sed -i 's@"$ref": "#/definitions/'"$ref"'"@'"$contents"'@g' {}
```

Reply:
>Use -exec on find, rather than xargs (


find . -maxdepth 1 -name '*.json' -exec bash -c 'sed -i "s@\"\$ref\": \"#/definitions/$1\"@$2@g" "$0"' {} "$ref" "$contents" \;

received the same error - ./export-openapi.sh: line 87: /usr/bin/find: Argument list too long

Reply:
I have included all the sed/awk commands I've tried so far, along with any associated errors - https://gist.github.com/Kalaiselvi84/9e94d55cb84c93d80d12818f5f347a51

Reply:
I wonder if the contents being replaced it just too large for a single bash command (looking at you Pod Spec) - in which case, probably time to convert this to a Go program.

Reply:
working on it👍🏻

Reply:
`awk: not an option: -i` is because the make command is using `mawk`